internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -427,3 +428,30 @@ func stopSpinner(spinner *progress.Spinner, err error, label string) {
 	}
 	spinner.Stop(log.Serrorf("%s\n", label))
 }
+
+// upgradeStackUpdateErrorHandling is used to handle the error returned by CFN
+// stack update when upgrading the application/environment.
+func upgradeStackUpdateErrorHandling(name string, err error) (bool, error) {
+	var emptyChangeSet *cloudformation.ErrChangeSetEmpty
+	var alreadyInProgErr *cloudformation.ErrStackUpdateInProgress
+	var obsoleteChangeSetErr *cloudformation.ErrChangeSetNotExecutable
+	switch updateErr := err; {
+	case errors.As(updateErr, &emptyChangeSet):
+		// The changes are already applied, nothing to do. Exit successfully.
+		return false, nil
+	case errors.As(updateErr, &alreadyInProgErr):
+		// There is another update going on in the environment, retry the upgrade.
```

Comment:
Does the `alreadyInProgErr` not happen for app stack, or we need to add "application" in the comment as well? (I didn't put this as a suggested change because I really am not sure)